src/OpenTelemetry.Contrib.Instrumentation.Wcf/TelemetryServiceBehavior.cs

```diff
@@ -0,0 +1,66 @@
+﻿// <copyright file="TelemetryServiceBehavior.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.ServiceModel;
+using System.ServiceModel.Description;
+using System.ServiceModel.Dispatcher;
+
+namespace OpenTelemetry.Contrib.Instrumentation.Wcf
+{
+#if NETFRAMEWORK
+    /// <summary>
+    /// An <see cref="IServiceBehavior"/> implementation add the the
+    /// <see cref="TelemetryDispatchMessageInspector"/> to service operations.
+    /// </summary>
+    public class TelemetryServiceBehavior : IServiceBehavior
+    {
+        /// <inheritdoc />
+        public void AddBindingParameters(ServiceDescription serviceDescription, ServiceHostBase serviceHostBase, System.Collections.ObjectModel.Collection<ServiceEndpoint> endpoints, System.ServiceModel.Channels.BindingParameterCollection bindingParameters)
+        {
+        }
+
+        /// <inheritdoc/>
+        public void ApplyDispatchBehavior(ServiceDescription serviceDescription, ServiceHostBase serviceHostBase)
+        {
+            foreach (var channelDispatcherBase in serviceHostBase.ChannelDispatchers)
+            {
+                var channelDispatcher = (ChannelDispatcher)channelDispatcherBase;
+                foreach (var endpointDispatcher in channelDispatcher.Endpoints)
+                {
+                    var actionMappings = new Dictionary<string, ActionMetadata>(StringComparer.OrdinalIgnoreCase);
```

Comment:
This code looks similar to what is in [TelemetryEndpointBehavior](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/739c6b706fd1b5beacf9c7a8b5b1f0970b316370/src/OpenTelemetry.Contrib.Instrumentation.Wcf/TelemetryEndpointBehavior.cs#L65-L76) could we refactor into an internal static so both could share the same implementation?

Reply:
@CodeBlanch Makes sense yes - I'm assuming that is a separate PR given it affects other files. Will look at getting that done today.

Reply:
Done. Please refer to commit c6fae3639293e98bd887e2cda89df50ef11d6bae